### PR TITLE
Fix explanations about core/extended functionalities

### LINKED ISSUE #56: Clarify core/extended funcationality comment
The second paragraph of Sec 1 "Introduction" says:
> The ADQL specification makes no distinction between core and advanced or extended functionalities.

But this seems to be contradicted by other parts of the standard, e.g. the presence of section 4 "Optional components" and the comment in section 1.2 "Extended functionality":
> This document defines the minimum set of functions, operators and datatypes that a service MUST implement in order to register as a service that implements this version of the ADQL specification.

Is the second paragraph of the introduction a relic of earlier versions that ought to be reworded or removed?  Or have I misunderstood its intent?

### PR DESCRIPTION
**Pull request by gmantele**
This PullRequest fixes the explanations about the core, optional and extended features. This modification mostly applies to the Introduction section.

Fixes #56

### CODE REVIEW

**Review by mbtaylor — APPROVED** [review]
Changes look good.  My only query is the reference to BNF which I understand will be replaced by PEG(?) in future versions, but that doesn't relate to #56, so should be tackled separately.

**Comment by gmantele** [comment]
Indeed, BNF will be replaced by PEG. But that's not yet the case in this version. In the current text, replacing "BNF" by "PEG" should still work. Both aims for the same thing ; PEG is just reliably machine-readable compared to this BNF flavor.

File: ADQL.tex [hunk 1]
```diff
@@ -84,26 +84,22 @@ \section{Introduction}
...
 therefore this specification defines syntactical correctness only. Type safety
-has been achieved as far as it can be done in SQL. The exact meaning of keywords
-indicating requirement levels can be found in the References section.
-%Should this be 'Conformance-related definitions' not 'References' ?
+has been achieved as far as it can be done in SQL. 
+
+This specification provides a complete definition of the ADQL through a single
+Backus Naur Form (BNF) based language definition. It includes both mandatory
+and optional features. Thus, this BNF aims to help implementing ADQL core
+features as well as some or all of the defined optional ones.
```
Comment (Zarquan) [inline ADQL.tex:102]: Thus, this BNF aims to help implementing ADQL core features as well as ~~some or all of~~ the defined optional ones.

File: ADQL.tex [hunk 2]
```diff
@@ -126,11 +122,13 @@ \subsection{Extended functionality}
 % Explicit permisson for services that provide additional functionality.
 This document defines the minimum set of functions, operators and datatypes
 that a service MUST implement in order to register as a service that
-implements this version of the ADQL specification.
+implements this version of the ADQL specification. It also defines optional
+features that a service MUST implement as such if it requires some or all of
+them.
```
Comment (Zarquan) [inline ADQL.tex:127]: It also defines optional features that a service ~~MUST~~ **MAY**  implement ~~as such~~ if it requires ~~some or all of~~ them.

Comment (Zarquan) [reply ADQL.tex:127]: They aren't optional if it MUST implement them.

Comment (mbtaylor) [reply ADQL.tex:127]: I think the intention is that if they are implemented at all, they have to be implemented as described in this specification ("as such").

Comment (gmantele) [reply ADQL.tex:127]: Actually my "MUST" was on the "as such". I meant that if a feature (e.g. put a string in lower case) is desired, it MUST be implemented as specified in the ADQL document (e.g. the function LOWER). But maybe it should rather be a "SHOULD" or something like "it is strongly encouraged to implement the corresponding optional ADQL feature, if existing".

That may be too difficult to write (at least, it is for me, with my poor english :-S ). And this is probably asking too much as well :/

File: ADQL.tex (hunk 1 above)
Comment (gmantele) [reply ADQL.tex:102]: Ok. I'll fix that in the next commit/PullRequest.

File: ADQL.tex (hunk 2 above)
Comment (gmantele) [reply ADQL.tex:127]: > I think the intention is that if they are implemented at all, they have to be implemented as described in this specification ("as such").

Yep. That's what I meant :D

Comment (Zarquan) [reply ADQL.tex:127]: Ok, sorry, that wasn't clear to me. Suggest a long form to be explicit?
"If a service does provide one of the optional features, it MUST be implemented according to this specification."

**Pull request merged**